interfaces/mount/backend.go

```diff
@@ -79,7 +80,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 		return fmt.Errorf("cannot synchronize mount configuration files for snap %q: %s", snapName, err)
 	}
 	if err := UpdateSnapNamespace(snapName); err != nil {
-		return fmt.Errorf("cannot update mount namespace of snap %q: %s", snapName, err)
+		// try to discard the mount namespace and retry, but only if there aren't daemons in the snap
```

Comment:
should we consider whether they refresh-mode: endure as well?

Reply:
It was my original idea, but since you was worried by daemons in general, I put it that way. Should I then only fail if it's a daemon AND the refresh mode is "endure"?

Reply:
@pedronis So... should I replace the "if" with

    if app.Daemon != "" && app.RefreshMode == "endure" {

Reply:
yes, something like that

Reply:
Done.